### ISSUE TITLE
Clang false positive missing-designated-field-initializers

### ISSUE THREAD

**Issue by sabudilovskiy** [llvm/llvm-project#108979]
```cpp
#include <string>

struct Test {
    std::string a;
    std::string b;
};

int main() { (void)Test{.b = ""}; }
```

expected behaviour: ok
got behaviour: 
```cpp
missing field 'a' initializer [-Wmissing-designated-field-initializers]
    8 | int main() { (void)Test{.b = ""}; }
```

https://godbolt.org/z/hTa1vEo1M

**Comment by shafik** [comment]
Note gcc behaves the same: https://godbolt.org/z/rvcGfMsoK

this ends up being a duplicate of: https://github.com/llvm/llvm-project/issues/83070
